metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -276,6 +227,79 @@ func NewContainerMetadataEnricher(
 	return enricher
 }
 
+func getPodMetadataWatchers(config *kubernetesConfig, resource kubernetes.Resource, nodeScope bool) (kubernetes.Watcher, kubernetes.Watcher, kubernetes.Watcher) {
+	client, err := kubernetes.GetKubernetesClient(config.KubeConfig, config.KubeClientOptions)
+	if err != nil {
+		return nil, nil, nil
+	}
+
+	options := kubernetes.WatchOptions{
+		SyncTimeout: config.SyncPeriod,
+	}
+
+	log := logp.NewLogger(selector)
+
+	// Watch objects in the node only
+	if nodeScope {
+		nd := &kubernetes.DiscoverKubernetesNodeParams{
+			ConfigHost:  config.Host,
+			Client:      client,
+			IsInCluster: kubernetes.IsInCluster(config.KubeConfig),
+			HostUtils:   &kubernetes.DefaultDiscoveryUtils{},
+		}
+		options.Node, err = kubernetes.DiscoverKubernetesNode(log, nd)
+		if err != nil {
+			return nil, nil, nil
+		}
+	}
+
+	log.Debugf("Initializing a new Kubernetes watcher using host: %v", config.Host)
+
+	watcher, err := kubernetes.NewNamedWatcher("resource_metadata_enricher", client, resource, options, nil)
+	if err != nil {
+		logp.Err("Error initializing Kubernetes metadata enricher: %s", err)
+		return nil, nil, nil
+	}
+
+	if watcher == nil {
+		logp.Info("Kubernetes metricset enriching is disabled")
+		return nil, nil, nil
+	}
+
+	nodeWatcher, err := kubernetes.NewNamedWatcher("resource_metadata_enricher_node", client, &kubernetes.Node{}, options, nil)
+	if err != nil {
+		logp.Err("Error creating watcher for %T due to error %+v", &kubernetes.Node{}, err)
+		return nil, nil, nil
```

Comment:
In case of error of nodeWatcher or NamespaceWatcher why setting `watcher` also to nil? I would expect to keep the main resource `watcher` as is and set nil the ones that could not be initialised

Reply:
good catch, done - https://github.com/elastic/beats/pull/29133/commits/ec17b227c511988c8648073783dd2b3640515481